### PR TITLE
subclass support for GtkEntry

### PR DESCRIPTION
**Pull request by pbor**

### CODE REVIEW

**Review by bilelmoussaoui — COMMENTED** [review]
There are a bunch of other vfuncs, I guess you are planning to add those as well?

File: gtk/src/subclass/entry.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            }
+        }
+    }
+}
+
+unsafe impl<T: EntryImpl> IsSubclassable<T> for Entry {
+    fn class_init(class: &mut glib::Class<Self>) {
+        Self::parent_class_init::<T>(class);
+
+        if !crate::rt::is_initialized() {
```
Comment (GuillaumeGomez) [inline gtk/src/subclass/entry.rs:55]: Isn't it checked in the parent class already?

Comment (pbor) [reply gtk/src/subclass/entry.rs:55]: I just blindly copied from button :)

**Comment by pbor** [comment]
> There are a bunch of other vfuncs, I guess you are planning to add those as well?

I can add them, but a lot of the vfuncs are just action vfuncs related to moving with the keyboard... do we need them?

I copied from `button.rs` and I see that that one also just has a couple of methods will in C there are many more vfuncs...

File: gtk/src/subclass/entry.rs (hunk 1 above)
Comment (pbor) [reply gtk/src/subclass/entry.rs:55]: All the other modules seem to do the same, but it is true that this is checked in widget.rs... Maybe we should drop it everywhere, but in that case it should go in a separate MR

Comment (GuillaumeGomez) [reply gtk/src/subclass/entry.rs:55]: Sounds like a plan. I'll send a PR for that.

Comment (sdroege) [reply gtk/src/subclass/entry.rs:55]: Maybe first `git blame` why they were added... there's a reason.

Comment (GuillaumeGomez) [reply gtk/src/subclass/entry.rs:55]: I was the one adding this check, just didn't think it through: https://github.com/gtk-rs/gtk3-rs/commit/cb53101568b1332acb48494ab5d82588bc04bd77

Comment (sdroege) [reply gtk/src/subclass/entry.rs:55]: Apparently you didn't :) See https://github.com/gtk-rs/gtk3-rs/issues/543

I think this code is correct and needed.

Comment (GuillaumeGomez) [reply gtk/src/subclass/entry.rs:55]: The issue is apparently about non-widget items. Oh well...

Comment (sdroege) [reply gtk/src/subclass/entry.rs:55]: @bilelmoussaoui can probably tell us for sure

Comment (bilelmoussaoui) [reply gtk/src/subclass/entry.rs:55]: I did that to panic as early as possible

Comment (GuillaumeGomez) [reply gtk/src/subclass/entry.rs:55]: Shouldn't it need to be before `Self::parent_class_init::<T>(class);` then?

Comment (pbor) [reply gtk/src/subclass/entry.rs:55]: I see you opened a PR, so I will resolve the conversation here for now.

**Review by sdroege — APPROVED** [review]

**Pull request merged**